R/error-checks.R

```diff
@@ -48,11 +47,13 @@ validateEnumValue <- function(enum, value) {
 }
 
 typeNamesFrom <- function(type) {
-  if (is.character(type)) {
-    return(type)
-  }
   type <- c(type)
-  sapply(type, function(t) t$classname)
+  sapply(type, function(t){
+    if (is.character(t)){
+      return(t)
+    }
+    t$classname

```

Comment:
Otherwise it produces an error when mixing Class and strings (e.g. `validateIsOfType(Quantity, "character")`)